client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks/useStocktakeLineEdit.ts

```diff
@@ -35,6 +44,58 @@ export const useStocktakeLineEdit = (
       })
     );
 
+  const save = async () => {
+    let result;
+    try {
+      result = await mutateAsync(draftLines);
+    } catch (e) {
+      return { errors: errorMessage(e) };
+    }
+
+    const insertResults = result.batchStocktake?.insertStocktakeLines || [];
+    const updateResults = result.batchStocktake?.updateStocktakeLines || [];
+
+    let errorMessagesMap: { [key: string]: string } = {};
+
+    for (const { response, id } of [...insertResults, ...updateResults]) {
+      // First unset error
+      errors.unsetError(id);
+      // No error
+      if (response.__typename === 'StocktakeLineNode') continue;
+
+      let { error } = response;
```

Comment:
:nit not reassigned. Also, when destructuring - good to indicate the immutability of props taken
```suggestion
      const { error } = response;
```

Reply:
cfe481f9f56dabfd7dcef924b1bc70caf8a818b6, I think I am used to typescript complaining about this, do we have this switched of ? or is my TS misconfigured ?